File: day 01/day1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

public class AdventOfCode {
	public static void main(String[] args) throws FileNotFoundException {
		
		File file = new File("D:\\Java\\AdventOfCode\\in.txt");
		Scanner cin = new Scanner(file);
		Set<Integer> numbers = new HashSet<>();
		int n;
		while (cin.hasNextInt()) {
			n = cin.nextInt();
			if (numbers.contains(2020 - n)) {
				System.out.println("Part 1: " + n * (2020 - n));
			}
			numbers.add(n);
		}

		for (Integer i : numbers) {
			for (Integer j : numbers) {
				if (numbers.contains(2020 - i - j)) {
					System.out.println("Part 2: " + i * j * (2020 - i - j));
					System.exit(0);
				}
			}
		}
	}
}

File: day 01/day1.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace AdventProject
{
  class AdventOfCode
  {
    static void Main(string[] args)
    {
      HashSet<int> numbers = new HashSet<int>();

      string inputFile = Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.FullName + @"\in.txt";
      StreamReader reader = new StreamReader(inputFile);

      while (!reader.EndOfStream)
      {
        int n;
        if (int.TryParse(reader.ReadLine(), out n) && n != 0)
        {
          if (numbers.Contains(2020 - n))
          {
            Console.WriteLine("Part 1: " + n * (2020 - n));
          }
          numbers.Add(n);
        }
      }

      foreach (var i in numbers)
      {
        foreach (var j in numbers)
        {
          if (numbers.Contains(2020 - i - j))
          {
            Console.WriteLine("Part 2: " + i * j * (2020 - i - j));
            return;
          }
        }
      }
    }
  }
}
